### PR TITLE
Add argo workflow ID to slack notification

### PR DESCRIPTION
**Pull request by eparkmonster**
## Why
Want to add the argo workflow id to slack notification
[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1643)

## This PR

Updates the notification solid and sensor to pass the workflow id. Also updated the test.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py [hunk 1]
```diff
@@ -66,10 +67,12 @@ def base_notify_slack_of_egress_validation_results(
 ) -> str:
     gcp_env = context.solid_config["gcp_env"]
     dataset_name = context.solid_config["dataset_name"]
+    argo_id = context.solid_config["argo_id"]
 
     if validation_results.has_problems():
         message_lines = [
             f"Problems identified in post-validation for HCA {gcp_env} dataset {dataset_name}:",
+            f"Triggering Argo workflow_id: {argo_id}",
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py:75]: 🤔 It'd be useful to include this in the "has no problems" branch on line 81 below, to be able to do a similar tie out with successful validations + workflows.

File: orchestration/dagster_orchestration/hca_orchestration/sensors.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ def generate_run_request(self, workflow: ExtendedArgoWorkflow) -> RunRequest:
         inflated_workflow = workflow.inflate()
 
         dataset_name = inflated_workflow.params_dict()['data-repo-name'].removeprefix("datarepo_")
+        argo_id = inflated_workflow.metadata.uid
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/sensors.py:32]: ⛏️ I think we should call this `argo_workflow_id`, as that's the entity to which the identifier is attached.

**Review by aherbst-broad — COMMENTED** [review]
Looking good, had a few comments.

**Review by ThePletch — CHANGES_REQUESTED** [review]
let's make sure the argo workflow id is present in all notifications from this solid.

File: orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py [hunk 3]
```diff
@@ -66,16 +67,19 @@ def base_notify_slack_of_egress_validation_results(
...
             f"Problems identified in post-validation for HCA {gcp_env} dataset {dataset_name}:",
+            f"Triggering Argo workflow ID: {argo_workflow_id}",
             "Duplicate lines found: " + str(validation_results.duplicates),
             "Null file references found: " + str(validation_results.null_file_refs),
             "Dangling project references found: " + str(validation_results.dangling_project_refs)
         ]
     else:
-        message_lines = [f"HCA {gcp_env} dataset {dataset_name} has passed post-validation."]
+        message_lines = [
+            f"HCA {gcp_env} dataset {dataset_name} workflow ID {argo_workflow_id} has passed post-validation."]
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py:82]: I think we should follow the format from the problems branch and separate this into multiple lines, i.e., 

```suggestion
            f"HCA {gcp_env} dataset {dataset_name} has passed post-validation.",
            f"Argo Workflow ID: {argo_workflow_id}"
            ]
```

**Review by aherbst-broad — APPROVED** [review]
lgtm, had a suggestion about breaking the success path into multiple lines.

File: orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py (hunk 3 above)
Comment (ThePletch) [reply orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py:82]: agreed - this helps communicate that the workflow ID isn't coupled to the dataset, as well.

**Review by ThePletch — APPROVED** [review]
seconding drew's comment re: formatting the "passed" message, otherwise this looks lovely

**Pull request merged**